src/wrap_fn_ptr.rs

```diff
@@ -20,23 +20,43 @@ impl<F> WrappedFnPtr<F> {
 }
 
 macro_rules! wrap_fn_ptr {
```

Comment:
I'll add docs to explain all of that.  I'll be sure to add docs to any non-trivial/exported macro; I agree that's very important for macros since they're hard to read.

Reply:
I've added docs for `wrap_fn_ptr!` now, as well as `trait HasFnPtr`, `struct WrappedFnPtr`, and `bd_fn!`.